src/runtime_src/core/common/info_platform.cpp

```diff
@@ -137,7 +137,9 @@ add_controller_info(const xrt_core::device* device, ptree_type& pt)
     sc.add("version", xrt_core::device_query<xq::xmc_sc_version>(device));
     sc.add("expected_version", xrt_core::device_query<xq::expected_sc_version>(device));
     ptree_type cmc;
-    cmc.add("version", xrt_core::device_query<xq::xmc_version>(device));
+    std::stringstream version;
+    version << "0x" << std::hex << std::stoi(xrt_core::device_query<xq::xmc_version>(device)); 
```

Comment:
1. Why aren't we using `ver = std::stoul(ver, nullptr, 16);` to convert to hex?
2. what if std::stoi fails? Please add a try-catch around it